src/xdocs/checks/whitespace/whitespaceafter.xml

```diff
@@ -122,88 +122,96 @@
       </subsection>
 
       <subsection name="Examples" id="Examples">
-        <p>
+        <p id="Example1-config">
```

Comment:
Why do we need these IDs?

Reply:
Seems this is missing from https://github.com/checkstyle/checkstyle/pull/13357#issue-1787842401 and other things like changes to the example input file.

Reply:
It to generate link to examples, there is separate PR that allows link image popup for copy. 
Very useful. @stoyanK7 , decided to do them from the beginning.

Reply:
You're right. I didn't add it there but explained it in https://github.com/checkstyle/checkstyle/pull/13357#discussion_r1260221188. Adding it in description too.

Reply:
Thanks. Is there a reason the macro doesn't create the links itself so that we don't need to add them manually? I guess it would put the link too far below the text it is connected to.

Reply:
As far as I'm aware, the macro would not be able to do this - edit a paragraph outside of its scope.

If we really want to automate it - I imagine some method that runs before the Xdoc generation and edits the templates. That method iterates through the macros, constructs an id based on parameters of the macro(`Example2-code` for example), and adds it to the paragraph preceding the macro.

Reply:
I was thinking more of adding the link to the end of the paragraph right where we start the scope. I am good regardless, it is just something to think about.

Reply:
Even if the paragraph is a whole screen-height long, the anchor will appear at the end of it, right above the example snippet. That's how the javascript that appends the anchors works.

Or am I misunderstanding you?

Reply:
Let's continue this discussion in separate issue, we can always improve in separate PR.